cli/azd/extensions/azure.foundry.ai.agents/internal/project/parser.go

```diff
@@ -0,0 +1,1291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"azureaiagent/internal/pkg/agents/agent_yaml"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/appcontainers/armappcontainers"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cognitiveservices/armcognitiveservices"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armsubscriptions"
+	"github.com/azure/azure-dev/cli/azd/pkg/azdext"
+	"github.com/azure/azure-dev/cli/azd/pkg/graphsdk"
+	"github.com/google/uuid"
+)
+
+type FoundryParser struct {
+	AzdClient *azdext.AzdClient
+}
+
+// Check if there is a service using containerapp host and contains agent.yaml file in the service path
+func (p *FoundryParser) shouldAttach(ctx context.Context) (bool, error) {
+	projectClient := p.AzdClient.Project()
+	projectResponse, err := projectClient.Get(ctx, &azdext.EmptyRequest{})
+	if err != nil {
+		return false, fmt.Errorf("failed to get project: %w", err)
+	}
+	project := projectResponse.Project
+
+	projectPath := project.Path
+	for _, service := range project.Services {
+		if service.Host == "containerapp" {
```

Comment:
The service level registration would also avoid this check since you can filter to only bind to specific host types.